lib/vagrant-libvirt/config.rb

```diff
@@ -538,6 +538,15 @@ def _generate_uri
       end
 
       def finalize!
+        @uri = ENV['VAGRANT_LIBVIRT_URI'] if (@uri == UNSET_VALUE && ENV['VAGRANT_LIBVIRT_URI'])
+        if (@uri && @uri != UNSET_VALUE)
+          uri = URI(@uri)
+          @driver = u.scheme.sub('+ssh','') if @driver == UNSET_VALUE
+          @host = uri.host if @host == UNSET_VALUE
+          @connect_via_ssh = uri.scheme.include?("+ssh") if @connect_via_ssh == UNSET_VALUE
+          @username = uri.user if @username == UNSET_VALUE
+          @password = uri.password if @password == UNSET_VALUE
+        end
```

Comment:
This will prevent the generation of the `@uri` below, see line 566 any time VAGRANT_LIBVIRT_URI is defined, even if a Vagrantfile defined all the other pieces of information for `@driver`, `@host`, `@connect_via_ssh`, `@username` and `@password`.

At the same time it will fail to make sure that the individual values have been set based on the provided `@uri`. This seems like a potential for unexpected behaviour.

I would suggest that if the `ENV['VAGRANT_LIBVIRT_URI']` should provide the `@uri`, then all the other settings should be constructed from and replace any provided values. Or the `ENV['VAGRANT_LIBVIRT_URI']` value should be ignored if the other values have been specifically provided.

Also critically important, if using the environment variable and ignoring what has been set in the Vagrantfile, it seems pretty important to report this to user.

Putting this all in a separate method will also make it easier to add some tests to just check the variables relating to `@uri`, which I think a change like this will need.